src/main/python/keycodes/keycodes.py

```diff
@@ -904,5 +920,10 @@ def generate_keycodes_for_mask(label, description):
 
     recreate_keycodes()
 
+    # Hide keycodes where .requires_feature isn't supported by the keyboard.
+    for kc in KEYCODES:
+        if not kc.is_supported_by(keyboard):
+            kc.hidden = True
```

Comment:
I think that should be: `kc.hidden = not kc.is_supported_by(keyboard)` otherwise once it's hidden it'll never reappear (e.g. if you have multiple keyboards plugged in with different featuresets)

Reply:
You're right! Done.